dashboard/app/models/census/ap_school_code.rb

```diff
@@ -15,47 +15,60 @@
 #
 
 class Census::ApSchoolCode < ApplicationRecord
-  self.primary_key = :school_code
+  self.primary_keys = :school_code, :school_year
 
   belongs_to :school, required: true
-  has_many :ap_cs_offering, foreign_key: :school_code, primary_key: :school_code
+  has_many :ap_cs_offering, foreign_key: [:school_code, :school_year], primary_key: [:school_code, :school_year]
 
   validates :school_code, presence: true, length: {is: 6}, format: {with: /\A[0-9]+\z/, message: "only allows numbers"}
 
+  validates :school_year, presence: true, numericality: {greater_than_or_equal_to: 2016, less_than_or_equal_to: 2030}
+
   # school_code is a 6-character string but in the input files we may have treated
   # them as integers and cut off leading zeros. Add them back if necessary.
   def self.normalize_school_code(raw_school_code)
     format("%06d", raw_school_code.to_i)
   end
 
-  def self.seed_from_csv(filename)
+  def self.construct_object_key(school_year)
+    "ap_school_codes/#{school_year}-#{school_year + 1}.csv"
+  end
+
+  def self.seed_from_csv(filename, school_year)
     ActiveRecord::Base.transaction do
```

Comment:
I know this functionality isn’t changing from what already existed, but I don’t think there’s any benefit to wrapping the CSV import in a transaction.  Do we really want to ROLLBACK all of the INSERTed rows if a single row encounters an error?  There’s a likely small database performance penalty, as well, for carrying out all of the INSERTs in a single transaction.